test/oauth.js

```diff
@@ -23,41 +22,87 @@ describe('oauth', function() {
     })
   })
 
-  describe('refreshAccessToken', function() {
-    it('should return (and refresh) an accessToken, given a refreshToken - constructor', function() {
-      const params = {
-        clientId: process.env.clientId,
-        clientSecret: process.env.clientSecret,
-        redirectUri: process.env.redirectUri,
-        refreshToken: process.env.refreshToken,
-      }
-      const hubspot = new Hubspot(params)
-      if (!process.env.refreshToken) {
-        return
-      } // hard to reproduce on CI. local testing only for now
-      return hubspot.oauth.refreshAccessToken().then(res => {
-        expect(res.refresh_token).to.equal(params.refreshToken)
-        expect(res).to.have.a.property('access_token')
-        expect(hubspot.auth.bearer).to.equal(res.access_token)
+  describe('getAccessToken', function() {
+    const code = 'a_fake_code'
+    const clientProperties = {
+      clientId: 'fake_client_id',
+      clientSecret: 'fake_client_secret',
+      redirectUri: 'some-redirect-uri',
+    }
+    const expectedResponse = {
+      refresh_token: 'a_fake_refresh_token',
+      access_token: 'a_fake_access_token',
+      expires_in: 21600,
+    }
+    const oauthTokenEndpoint = {
+      path: '/oauth/v1/token',
+      request: {
+        grant_type: 'authorization_code',
+        client_id: clientProperties.clientId,
+        client_secret: clientProperties.clientSecret,
+        redirect_uri: clientProperties.redirectUri,
+        code,
+      },
+      response: expectedResponse,
+    }
+    fakeHubspotApi.setupServer({ postEndpoints: [oauthTokenEndpoint] })
+
+    if (process.env.NOCK_OFF) {
+      it('should return a token from hubspot')
+    } else {
+      it('should return a token from hubspot', function() {
+        if (process.env.NOCK_OFF) {
+          console.warn(
+            'we currently do not test oauth with NOCK_OFF set to true',
```

Comment:
Nice 👍

Reply:
Yes but it looks terrible in the console so I've removed it.